tests/serve_request.rs

```diff
@@ -69,6 +70,31 @@ fn serves_requests_with_non_default_port(tmpdir: TempDir, port: u16) -> Result<(
     Ok(())
 }
 
+#[rstest]
+fn serves_requests_with_randomly_assigned_port(tmpdir: TempDir) -> Result<(), Error> {
+    let mut child = Command::cargo_bin("miniserve")?
+        .arg(tmpdir.path())
+        .arg("-p")
+        .arg("0".to_string())
+        .stdout(Stdio::piped())
+        .spawn()?;
+
+    sleep(Duration::from_secs(1));
+    child.kill()?;
+
+    let output = child.wait_with_output().expect("Failed to read stdout");
+    let all_text = String::from_utf8(output.stdout)?;
+
+    let re = Regex::new(r"http://127.0.0.1:(\d+)").unwrap();
+    let caps = re.captures(all_text.as_str()).unwrap();
+    let port_num = caps.get(1).unwrap().as_str().parse::<u32>().unwrap();
```

Comment:
Shouldn't this be parsed as a `u16` in which case the assert down below can be omitted because that's exactly the `u16` value range?

Reply:
For sure; updated!